R/data-storage.R

```diff
@@ -96,37 +92,90 @@ DataStorage <- R6::R6Class( # nolint object_name_linter
         ))
       })
     },
-    generate_session_id = function() {
-      paste(
-        c(
-          sample(c(letters, LETTERS, 0:9), 10),
-          format(Sys.time(), "%Y%d%m%H%M%S")
-        ),
-        collapse = ""
-      )
-    },
+
     close_connection = function() {
       rlang::abort("Method not implemented.")
     },
-    insert_checks = function(values, insert_time, bucket) {
-      checkmate::assert_string(bucket)
-      checkmate::assert_list(values)
-      checkmate::assert_flag(insert_time)
 
-      if (!insert_time) {
-        return(values)
-      }
+    write = function(values, bucket) {
+      rlang::abort("Method not implemented.")
+    },
 
-      if ("time" %in% names(values)) {
-        rlang::abort(paste0(
-          "You must not pass 'time' value into database.",
-          " It is set automatically."
-        ))
-      }
+    read_data = function(date_from, date_to, bucket) {
+      rlang::abort("Method not implemented.")
+    },
 
-      values$time <- as.character(Sys.time())
+    insert_checks = function(app_name, type, session, details, time) {
```

Comment:
Explicit arguments instead of passing a list.

This makes it much cleaner as we don't need `force` params or `insert_time`